modos/api.py

```diff
@@ -65,29 +68,17 @@ def __init__(
         has_assay: List = [],
         source_uri: Optional[str] = None,
     ):
-        self.s3_endpoint = s3_endpoint
+        self.htsget_endpoint = htsget_endpoint
+
         if s3_endpoint and not htsget_endpoint:
             htsget_endpoint = re.sub(r"s3$", "htsget", s3_endpoint)
-        self.htsget_endpoint = htsget_endpoint
-        self.path = Path(path)
+
         if s3_endpoint:
-            s3_opts = s3_kwargs or {"anon": True}
-            fs = s3fs.S3FileSystem(endpoint_url=s3_endpoint, **s3_opts)
-            if fs.exists(str(self.path / "data.zarr")):
-                zarr_s3_opts = s3_opts | {"endpoint_url": s3_endpoint}
-                self.archive = zarr.convenience.open(
-                    f"s3://{path}/data.zarr",
-                    storage_options=zarr_s3_opts,
-                )
-                return
+            self.storage = S3Storage(path, s3_endpoint, s3_kwargs)
         else:
-            fs = None
+            self.storage = LocalStorage(path)
         # Opening existing object
-        if (self.path / "data.zarr").exists():
-            self.archive = zarr.convenience.open(str(self.path / "data.zarr"))
-        # Creating from scratch
-        else:
-            self.archive = init_zarr(self.path, fs)
+        if not "id" in self.zarr.attrs.keys():
```

Comment:
__Question__: I replaced the `storage.exists()` method here compared to our first draft, because `ZARR_ROOT` will always exists after initializing the storage. Maybe you have a better/cleaner solution than checking for the attribute keys here?

Reply:
Ah right! then that makes the [current] Storage.exists method useless...

Perhaps we could do exactly that in `storage.exists()` and call it here? or even just `return self.zarr.attrs.keys()`

Reply:
Alternativerly, the method could be `storage.empty()` and return `not self.zarr.attrs.keys()`. This may be semantically more accurate.

Reply:
added `Storage.empty() -> bool` directly implemented in the abstract base class:

https://github.com/sdsc-ordes/modos-api/blob/refactor/remote/modos/storage.py#L43-L44

It is called in `MODO.__init__()` to be more explicit (see https://github.com/sdsc-ordes/modos-api/pull/68/commits/e7c3ffe63d474db40b94ed5f8c9336dd36c1c4bb)